libs/traits/src/lib.rs

```diff
@@ -125,6 +128,47 @@ pub trait PoolInspect<AccountId, CurrencyId> {
 	) -> Option<PriceValue<CurrencyId, Self::Rate, Self::Moment>>;
 }
 
+/// Variants for valid Pool updates to send out as events
+#[derive(Encode, Decode, Clone, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub enum UpdateState {
+	NoExecution,
+	Executed,
+	Stored,
+}
+
+/// A trait that supports modifications of pools
+pub trait PoolMutate<
+	AccountId,
+	Balance,
+	PoolId,
+	CurrencyId,
+	Rate,
+	MaxTokenNameLength: Get<u32>,
+	MaxTokenSymbolLength: Get<u32>,
+	MaxTranches: Get<u32>,
+>
+{
+	type TrancheInput: Encode + Decode + Clone + TypeInfo + Debug + PartialEq;
+	type PoolChanges: Encode + Decode + Clone + TypeInfo + Debug + PartialEq;
+
+	fn create(
+		admin: AccountId,
+		depositor: AccountId,
+		pool_id: PoolId,
+		tranche_inputs: Vec<Self::TrancheInput>,
+		currency: CurrencyId,
+		max_reserve: Balance,
+		metadata: Option<Vec<u8>>,
+	) -> DispatchResult;
+
+	fn update(
+		pool_id: PoolId,
+		changes: Self::PoolChanges,
+	) -> Result<(UpdateState, PostDispatchInfo), DispatchErrorWithPostInfo>;
```

Comment:
This return type scares me, but as long as it's not too awkward to work with it's fine....